core-bundle/src/Event/ContaoCoreEvents.php

```diff
@@ -15,7 +15,16 @@
 final class ContaoCoreEvents
 {
     /**
-     * The contao.backend_menu_build event is triggered when the backend menu is built.
+     * The contao.backend_header_menu_build event is triggered when the back end header menu is built.
+     *
+     * @var string
+     *
+     * @see MenuEvent
+     */
+    public const BACKEND_HEADER_MENU_BUILD = 'contao.backend_header_menu_build';
```

Comment:
As this is for Contao 4.9 only, we should actually not use an event name, but a different event class, shouldn't we? If not, I would still suggest to simply name this `BACKEND_HEADER_BUILD` (and shortname accordingly)

Reply:
You suggestion shows exactly why using two different events is not good in this case. 😄 It would simply be the `BackendMenuBuilder::class` event if we were able to use the same event in both methods.

But since we cannot, I guess we have to stick with `BACKEND_HEADER_MENU_BUILD`. Because `BACKEND_HEADER_BUILD` is misleading – after all, the event is triggered only when the back end header menu is build, not when the back end header is build.

Reply:
> You suggestion shows exactly why using two different events is not good in this case. 😄 It would simply be the BackendMenuBuilder::class event if we were able to use the same event in both methods.

I know we can use the same event, but we don't have to 😉 

> But since we cannot, I guess we have to stick with BACKEND_HEADER_MENU_BUILD. Because BACKEND_HEADER_BUILD is misleading – after all, the event is triggered only when the back end header menu is build, not when the back end header is build.

I don't find it misleading. It's a KnpMenu Event, so it's obviously about the menu regardless of the event name. But either is fine with me 😊